js/modules/k6/experimental/fs/module.go

```diff
@@ -0,0 +1,326 @@
+	// the current working directory (the k6 command is called from).
+	//
+	// This is done on purpose, although it diverges in some respect with
+	// how files are handled in different k6 contexts, so that we cater to
+	// and intuitive user experience.
+	//
+	// See #2781 and #2674.
+	path = fsext.Abs(initEnv.CWD.Path, path)
+
+	fs, ok := initEnv.FileSystems["file"]
+	if !ok {
+		common.Throw(mi.vu.Runtime(), errors.New("open() failed; reason: unable to access the file system"))
+	}
+
+	if exists, err := fsext.Exists(fs, path); err != nil {
+		return nil, fmt.Errorf("open() failed, unable to verify if %q exists; reason: %w", path, err)
+	} else if !exists {
+		return nil, newFsError(NotFoundError, fmt.Sprintf("no such file or directory %q", path))
+	}
+
+	if isDir, err := fsext.IsDir(fs, path); err != nil {
+		return nil, fmt.Errorf("open() failed, unable to verify if %q is a directory; reason: %w", path, err)
+	} else if isDir {
+		return nil, newFsError(
+			InvalidResourceError,
+			fmt.Sprintf("cannot open %q: opening a directory is not supported", path),
+		)
+	}
+
+	data, err := mi.cache.open(path, fs)
+	if err != nil {
+		return nil, err
+	}
+
+	return &File{
+		Path: path,
+		file: file{
+			path: path,
+			data: data,
+		},
+		vu:       mi.vu,
+		registry: mi.cache,
+	}, nil
+}
+
+// File represents a file and exposes methods to interact with it.
+//
+// It is a wrapper around the [file] struct, which is meant to be directly
+// exposed to the JS runtime.
+type File struct {
+	// Path holds the name of the file, as presented to [Open].
+	Path string `json:"path"`
+
+	// file contains the actual implementation for the file system.
+	file
+
+	// vu holds a reference to the VU this file is associated with.
+	//
+	// We need this to be able to access the VU's runtime, and produce
+	// promises that are handled by the VU's runtime.
+	vu modules.VU
+
+	// registry holds a pointer to the file registry this file is associated
+	// with. That way we are able to close the file when it's not needed
+	// anymore.
+	registry *cache
+}
+
+// Stat returns a promise that will resolve to a [FileInfo] instance describing
+// the file.
+func (f *File) Stat() *goja.Promise {
+	promise, resolve, _ := promises.New(f.vu)
+
+	go func() {
+		resolve(f.file.stat())
+	}()
+
+	return promise
+}
+
+// Read the file's content, and writes it into the provided Uint8Array.
+//
+// Resolves to either the number of bytes read during the operation
+// or EOF (null) if there was nothing more to read.
+//
+// It is possible for a read to successfully return with 0 bytes.
+// This does not indicate EOF.
+func (f *File) Read(into goja.Value) *goja.Promise {
+	promise, resolve, reject := f.vu.Runtime().NewPromise()
+
+	if common.IsNullish(into) {
+		reject(newFsError(TypeError, "read() failed; reason: into argument cannot be null or undefined"))
+		return promise
+	}
+
+	// We expect the into argument to be a `Uint8Array` instance
```

Comment:
This requirement is already a part of the method description. Do we still need to have this comment here?